cache/storage_redis.go

```diff
@@ -0,0 +1,82 @@
+package cache
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/garyburd/redigo/redis"
+)
+
+// RedisStorage implements Storage
+type RedisStorage struct {
+	// Pool is the redis connection pool (required)
+	Pool *redis.Pool
+
+	// TTL is the max TTL for cache items (required)
+	TTL time.Duration
+
+	// calculated version of TTL
+	ttlInSeconds int64
+	ttlOnce      sync.Once
+}
+
+// Get implements Storage
+func (r *RedisStorage) Get(ctx context.Context, key string) ([]byte, error) {
+	select {
+	case resp := <-r.asyncDo("GET", key):
```

Comment:
Per the comment on the method, it was my way of adding context support to a library that did not include it.
By extension allowing context timeout/cancel to shortcut the GET/SET even if we cannot actually stop it.

This makes me wonder if these calls should be instead wrapped in a circuit breaker.  This would essentially add context support as well as the other CB benefits